packages/components/src/presentation-components/tree/controlled/UseHierarchyLevelFiltering.ts

```diff
@@ -27,25 +27,29 @@ export interface UseHierarchyLevelFilteringProps {
  */
 export function useHierarchyLevelFiltering(props: UseHierarchyLevelFilteringProps) {
   const { nodeLoader, modelSource } = props;
+  const ongoingSubscriptions = useRef(new Map<string, Subscription>());
 
-  const applyFilter = useCallback(
-    (node: TreeNodeItem, info: PresentationInstanceFilterInfo) => {
-      applyHierarchyLevelFilter(nodeLoader, modelSource, node.id, info);
+  const handleFilterAction = useCallback(
```

Comment:
This `useCallback` is now useless because this function now returns new `applyFilter` and `clearFilter` functions. Either wrap them in `useCallback` or remove this `useCallback` usage.

Reply:
removed